### PR TITLE
[FSDE-12497] Refactored Data Table Bulk Actions

### PR DESCRIPTION
**Pull request by janibolkvadze**
### Commit summary

- [x] Addressed my comments on pr #241 
- [x] Fixed the additional issues I found:
        1. Sub-row selection - (Flow: Select a sub-row, select Show selected only, select another row. Does not filter the table to 1 row).
        2. `DataTable.tsx` `showSelectedOnly()` method inconsistent behavior. Needs to be rewritten.
        3. Select All inconsistent behavior with Bulk actions - Select `X` rows, select Show selected only, Deselect the SelectAll checkbox, only `X` rows are visible. The user is unable to see the other rows (initial state of Data Table).
        4. Select All not working on mobile.
        5. Visual issues on mobile - Layering - bottom gets covered with other elements, top bar height is inconsistent and links need to be changed by buttons.
        6. Fix Sorting Show Selected only mode
        7. Add support of Bulk Actions in Server Side mode
        8. Change Toolbar code snippets.

### CODE REVIEW

**Comment by SCMAUTO** [comment]
The CI pipeline run successfully in https://jenkinsprod.corp.intranet:8443/job/UX-CHI/job/Productive/job/Chi/job/PR-283/1/. :white_check_mark:

**Comment by SCMAUTO** [comment]
You can check this PRs instance in https://806cd40740a7a1a3d1704697792f281e5ac40666-chi-pr.enterprise.ctl.io (external) and in https://nginx-pr-283-ux-chi.kubeodc-test.corp.intranet/ (internal)

**Comment by SCMAUTO** [comment]
The CI pipeline run successfully in https://jenkinsprod.corp.intranet:8443/job/UX-CHI/job/Productive/job/Chi/job/PR-283/2/. :white_check_mark:

**Comment by SCMAUTO** [comment]
You can check this PRs instance in https://806cd40740a7a1a3d1704697792f281e5ac40666-chi-pr.enterprise.ctl.io (external) and in https://nginx-pr-283-ux-chi.kubeodc-test.corp.intranet/ (internal)

File: src/chi-vue/src/components/data-table-bulk-actions/DataTableBulkActions.tsx [hunk 1]
```diff
@@ -1,4 +1,4 @@
-import { Component, Prop, Vue } from 'vue-property-decorator';
+import { Component, Prop, Vue, Watch } from 'vue-property-decorator';
 import { findComponent } from '@/utils/utils';
```
Comment (evansito) [inline src/chi-vue/src/components/data-table-bulk-actions/DataTableBulkActions.tsx:2]: i was playing around in the complex example and i think i found some bugs / it's not working correctly :S

**Comment by evansito** [comment]
i was playing around in the complex example and i think i found some bugs / it's not working correctly :S

**Comment by janibolkvadze** [comment]
> i was playing around in the complex example and i think i found some bugs / it's not working correctly :S

Complex example does not currently use DataTableBulkActions..
Please share the bugs you are facing. If it's sorting-related, I'm currently working on changes to make another commit on this PR.

**Comment by evansito** [comment]
You're right! It's in regards to the show selected only it seems. E.g.:

image

image

**Comment by SCMAUTO** [comment]
The CI pipeline did not run successfully in https://jenkinsprod.corp.intranet:8443/job/UX-CHI/job/Productive/job/Chi/job/PR-283/3/. :x:

**Comment by SCMAUTO** [comment]
The CI pipeline run successfully in https://jenkinsprod.corp.intranet:8443/job/UX-CHI/job/Productive/job/Chi/job/PR-283/4/. :white_check_mark:

**Comment by SCMAUTO** [comment]
You can check this PRs instance in https://6b066f105edd8fa3ed37a04939c82321cec3fdae-chi-pr.enterprise.ctl.io (external) and in https://nginx-pr-283-ux-chi.kubeodc-test.corp.intranet/ (internal)

**Comment by SCMAUTO** [comment]
The CI pipeline run successfully in https://jenkinsprod.corp.intranet:8443/job/UX-CHI/job/Productive/job/Chi/job/PR-283/5/. :white_check_mark:

**Comment by SCMAUTO** [comment]
You can check this PRs instance in https://b3b3d186aa2a1ea17331153f0da8df82d6c0a271-chi-pr.enterprise.ctl.io (external) and in https://nginx-pr-283-ux-chi.kubeodc-test.corp.intranet/ (internal)

**Comment by SCMAUTO** [comment]
The CI pipeline run successfully in https://jenkinsprod.corp.intranet:8443/job/UX-CHI/job/Productive/job/Chi/job/PR-283/6/. :white_check_mark:

**Comment by SCMAUTO** [comment]
You can check this PRs instance in https://9ed15568580797f4efb6217de4b1d2392ccc4ed5-chi-pr.enterprise.ctl.io (external) and in https://nginx-pr-283-ux-chi.kubeodc-test.corp.intranet/ (internal)

**Comment by SCMAUTO** [comment]
The CI pipeline run successfully in https://jenkinsprod.corp.intranet:8443/job/UX-CHI/job/Productive/job/Chi/job/PR-283/8/. :white_check_mark:

**Comment by SCMAUTO** [comment]
You can check this PRs instance in https://ef83c0692a2a004c5e810cea05720cddb8960678-chi-pr.enterprise.ctl.io (external) and in https://nginx-pr-283-ux-chi.kubeodc-test.corp.intranet/ (internal)

**Comment by SCMAUTO** [comment]
The CI pipeline run successfully in https://jenkinsprod.corp.intranet:8443/job/UX-CHI/job/Productive/job/Chi/job/PR-283/9/. :white_check_mark:

**Comment by SCMAUTO** [comment]
The CI pipeline run successfully in https://jenkinsprod.corp.intranet:8443/job/UX-CHI/job/Productive/job/Chi/job/PR-283/10/. :white_check_mark:

**Comment by SCMAUTO** [comment]
You can check this PRs instance in https://3c7bd345c74230e279e37ed548fab48108f3450e-chi-pr.enterprise.ctl.io (external) and in https://nginx-pr-283-ux-chi.kubeodc-test.corp.intranet/ (internal)

**Comment by SCMAUTO** [comment]
The CI pipeline run successfully in https://jenkinsprod.corp.intranet:8443/job/UX-CHI/job/Productive/job/Chi/job/PR-283/11/. :white_check_mark:

**Comment by SCMAUTO** [comment]
You can check this PRs instance in https://b53ec720c808793c7a1ff006f88227589b48a104-chi-pr.enterprise.ctl.io (external) and in https://nginx-pr-283-ux-chi.kubeodc-test.corp.intranet/ (internal)

**Comment by SCMAUTO** [comment]
The CI pipeline did not run successfully in https://jenkinsprod.corp.intranet:8443/job/UX-CHI/job/Productive/job/Chi/job/PR-283/12/. :x:

**Comment by SCMAUTO** [comment]
The CI pipeline run successfully in https://jenkinsprod.corp.intranet:8443/job/UX-CHI/job/Productive/job/Chi/job/PR-283/13/. :white_check_mark:

**Comment by SCMAUTO** [comment]
You can check this PRs instance in https://7e9e8f2767049accf9ca9f6788880a2417e1b77e-chi-pr.enterprise.ctl.io (external) and in https://nginx-pr-283-ux-chi.kubeodc-test.corp.intranet/ (internal)

**Comment by SCMAUTO** [comment]
The CI pipeline run successfully in https://jenkinsprod.corp.intranet:8443/job/UX-CHI/job/Productive/job/Chi/job/PR-283/15/. :white_check_mark:

**Comment by SCMAUTO** [comment]
You can check this PRs instance in https://8ad426717121eeaa2778ff55167b919a16c54b55-chi-pr.enterprise.ctl.io (external) and in https://nginx-pr-283-ux-chi.kubeodc-test.corp.intranet/ (internal)

**Comment by SCMAUTO** [comment]
The CI pipeline run successfully in https://jenkinsprod.corp.intranet:8443/job/UX-CHI/job/Productive/job/Chi/job/PR-283/16/. :white_check_mark:

**Comment by SCMAUTO** [comment]
You can check this PRs instance in https://9421f4224848a5b06b9828c4f44f00a0ffa785a0-chi-pr.enterprise.ctl.io (external) and in https://nginx-pr-283-ux-chi.kubeodc-test.corp.intranet/ (internal)

**Comment by SCMAUTO** [comment]
The CI pipeline run successfully in https://jenkinsprod.corp.intranet:8443/job/UX-CHI/job/Productive/job/Chi/job/PR-283/17/. :white_check_mark:

**Comment by SCMAUTO** [comment]
You can check this PRs instance in https://c240eb28268f25adb7250648b342fae8369b609e-chi-pr.enterprise.ctl.io (external) and in https://nginx-pr-283-ux-chi.kubeodc-test.corp.intranet/ (internal)

File: src/chi-vue/src/components/data-table/DataTable.tsx [hunk 2]
```diff
@@ -1252,14 +1288,14 @@ export default class DataTable extends Vue {
     const classes = this._dataTableClasses(this.config.style, this._sortable),
       head = this._head(),
       toolbar = this._toolbar(),
-      bulkaction = this._bulkAction(),
+      bulkactions = this._bulkActions(),
```
Comment (evansito) [inline src/chi-vue/src/components/data-table/DataTable.tsx:1291]: any reason this isn't named bulkActions?

**Comment by SCMAUTO** [comment]
The CI pipeline run successfully in https://jenkinsprod.corp.intranet:8443/job/UX-CHI/job/Productive/job/Chi/job/PR-283/18/. :white_check_mark:

**Comment by SCMAUTO** [comment]
You can check this PRs instance in https://bf28ea02b7d10e83768e2b59885ea68d2f4da57c-chi-pr.enterprise.ctl.io (external) and in https://nginx-pr-283-ux-chi.kubeodc-test.corp.intranet/ (internal)

File: src/chi-vue/src/components/data-table/DataTable.tsx (hunk 2 above)
Comment (janibolkvadze) [inline src/chi-vue/src/components/data-table/DataTable.tsx:1291]:
```suggestion
      bulkActions = this._bulkActions(),
```

File: src/chi-vue/src/components/data-table/DataTable.tsx [hunk 3]
```diff
@@ -1252,14 +1288,14 @@ export default class DataTable extends Vue {
...
       body = this._body(),
       pagination = this._pagination();
 
     return (
       <div class={classes} role="table" ref="dataTable" data-table-number={dataTableNumber}>
         {toolbar}
-        {bulkaction}
+        {bulkactions}
```
Comment (janibolkvadze) [inline src/chi-vue/src/components/data-table/DataTable.tsx:1298]:
```suggestion
        {bulkActions}
```

**Review by josecarlosgonzalezv — APPROVED** [review]

**Review by jllr — APPROVED** [review]

**Review by evansito — APPROVED** [review]

**Review by evansito — APPROVED** [review]

**Comment by SCMAUTO** [comment]
The CI pipeline run successfully in https://jenkinsprod.corp.intranet:8443/job/UX-CHI/job/Productive/job/Chi/job/PR-283/19/. :white_check_mark:

**Comment by SCMAUTO** [comment]
You can check this PRs instance in https://d51f638c2dac2c18701f6bf549baff84443d6d5b-chi-pr.enterprise.ctl.io (external) and in https://nginx-pr-283-ux-chi.kubeodc-test.corp.intranet/ (internal)

**Review by dani-cl-madrid — APPROVED** [review]

**Pull request merged**